src/main/java/org/folio/service/pieces/PiecesClaimingService.java

```diff
@@ -196,39 +195,39 @@ private Future<ClaimingResults> createJobsByVendor(ClaimingCollection claimingCo
+    var updatePiecesFutures = new ArrayList<Future<List<String>>>();
+    pieceIdsByVendor.forEach((vendor, pieceIds) -> updatePiecesFutures.add(updatePieces(claimingCollection, vendor.getId(), pieceIds, requestContext).map(pieceIds)));
+    log.info("updatePiecesPerVendor:: Updating pieces per vendor, total count: {}", updatePiecesFutures.size());
+    return collectResultsOnSuccess(updatePiecesFutures);
+  }
+
+  private Future<List<Void>> createJobsPerVendorPerIntegrationDetail(JsonObject config, Map<Organization, List<String>> pieceIdsByVendor,
+                                                                     RequestContext requestContext) {
+    var createJobFutures = new ArrayList<Future<Void>>();
     pieceIdsByVendor.forEach((vendor, pieceIds) -> config.stream()
       .filter(configEntry -> isExportTypeClaimsAndCorrectVendorId(vendor.getId(), configEntry))
-      .forEach(configEntry -> {
-        log.info("createUpdatePiecesAndJobFutures:: Preparing job integration detail for vendor, vendor id: {}, pieces: {}, job key: {}",
-          vendor.getId(), pieceIds.size(), configEntry.getKey());
-        updatePiecesAndJobFutures.add(updatePiecesAndCreateJob(claimingCollection, pieceIds, configEntry, requestContext).map(pieceIds));
-      }));
-    return updatePiecesAndJobFutures;
+      .forEach(configEntry -> createJobFutures.add(createJob(configEntry.getKey(), configEntry.getValue(), pieceIds, requestContext))));
+    log.info("createJobsPerVendorPerIntegrationDetail:: Creating jobs vendor per integration details, total count: {}", createJobFutures.size());
+    return collectResultsOnSuccess(createJobFutures);
   }
 
-  private Future<Void> updatePiecesAndCreateJob(ClaimingCollection claimingCollection, List<String> pieceIds,
-                                                Map.Entry<String, Object> configEntry, RequestContext requestContext) {
-    log.info("updatePiecesAndCreateJob:: Updating pieces and creating a job, job key: {}, count: {}", configEntry.getKey(), pieceIds.size());
-    return createJob(configEntry.getKey(), configEntry.getValue(), pieceIds, requestContext).map(pieceIds)
-      .compose(v -> pieceUpdateFlowManager.updatePiecesStatuses(pieceIds, PieceBatchStatusCollection.ReceivingStatus.CLAIM_SENT,
-        claimingCollection.getClaimingInterval(), claimingCollection.getInternalNote(), claimingCollection.getExternalNote(), requestContext));
+  private Future<Void> updatePieces(ClaimingCollection claimingCollection, String vendorId, List<String> pieceIds, RequestContext requestContext) {
+    log.info("updatePieces:: Updating pieces for vendor, vendorId: {}, pieces: {}", vendorId, pieceIds.size());
+    return pieceUpdateFlowManager.updatePiecesStatuses(pieceIds, PieceBatchStatusCollection.ReceivingStatus.CLAIM_SENT,
+      claimingCollection.getClaimingInterval(), claimingCollection.getInternalNote(), claimingCollection.getExternalNote(), requestContext);
```

Comment:
if we can simplify this in scope of this pr, it would be good. we can create another using overload and method will take claimingCollection and extract to this method

Reply:
Maybe better to just extract 3 variables for visibility in this current `updatePiecesStatuses()` method, because `PieceUpdateFlowManager.updatePiecesStatuses()` doesn't really need to take ClaimingCollection as a parameter